pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -883,14 +889,18 @@ func (og *operationGenerator) GenerateMapVolumeFunc(
 			return volumeToMount.GenerateError("MapVolume failed", fmt.Errorf("Device path of the volume is empty"))
 		}
 
-		// When kubelet is containerized, devicePath may be a symlink at a place unavailable to
-		// kubelet, so evaluate it on the host and expect that it links to a device in /dev,
-		// which will be available to containerized kubelet. If still it does not exist,
-		// AttachFileDevice will fail. If kubelet is not containerized, eval it anyway.
-		mounter := og.GetVolumePluginMgr().Host.GetMounter(blockVolumePlugin.GetPluginName())
-		devicePath, err = mounter.EvalHostSymlinks(devicePath)
-		if err != nil {
-			return volumeToMount.GenerateError("MapVolume.EvalHostSymlinks failed", err)
+		// For in-tree drivers, actual device is attached either in WaitForAttach or SetUpDevice,
+		// not in MapDevice below, so devicePath could be checked here.
+		if !isCSI {
```

Comment:
@msau42 @wongma7 

Do you have any better ideas to handle this, so that operation_generator code doesn't need to care whether it is CSI?
We might make ```SetUpDevice``` for CSI return dummy device paht and make all driver resolve symlink from ```volumeMapPath``` below, but I'm not sure if it is better than current one.

Reply:
for csi, is the devicePath not the staging path returned by SetUpDevice (NodeStage)?